crates/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -309,8 +302,19 @@ impl<'block> BrilligBlock<'block> {
                         dfg.instruction_results(instruction_id)[0],
                         dfg,
                     );
-                    let param = self.convert_ssa_value(arguments[0], dfg);
-                    self.brillig_context.length_of_variable_instruction(param, result_register);
+                    let param_id = arguments[0];
+                    let param_variable = self.convert_ssa_value(param_id, dfg);
+
+                    self.brillig_context
+                        .length_of_variable_instruction(param_variable, result_register);
+
+                    let item_size = self.get_ssa_item_size(param_id, dfg);
+
+                    self.brillig_context.usize_op_in_place(
```

Comment:
This looks to be dividing the result register which contains the flattened size of the intrinsic result. Could we just combine `get_ssa_item_size` and `length_of_variable_instruction` to avoid the extra div instruction here?